components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/bulkimport/ExcelUserBulkImport.java

```diff
@@ -47,87 +53,90 @@ public ExcelUserBulkImport(BulkImportConfig config) {
+                userName = UserCoreUtil.addDomainToName(domainFreeName, userStoreDomain);
+            } else {
+                userName = UserCoreUtil.addDomainToName(userName, userStoreDomain);
+            }
 
-                if (userName != null && userName.trim().length() > 0) {
-                    try {
-                        if (!userStore.isExistingUser(userName)) {
-                            userStore.addUser(userName, null, null, null, null, true);
-                            success = true;
-                        } else {
-                            isDuplicate = true;
-                        }
-                    } catch (Exception e) {
+            if (userName != null && userName.trim().length() > 0) {
+                try {
+                    if (!userStore.isExistingUser(userName)) {
+                        userStore.addUser(userName, null, null, null, null, true);
+                        successCount++;
                         if (log.isDebugEnabled()) {
-                            log.debug(e);
+                            log.debug("User import successful - Username : " + userName);
                         }
-                        lastError = e.getMessage();
-                        fail = true;
+                    } else {
+                        duplicateCount++;
+                        duplicateUsers.add(userName);
+                        isDuplicate = true;
+                        log.error("User import unsuccessful - Username : " + userName + " - Error: Duplicate user");
+                        duplicateUsers.add(userName);
                     }
+                } catch (UserStoreException e) {
+                    fail = true;
+                    failCount++;
+                    log.error("User import unsuccessful - Username : " + userName + " - Error: " +
+                            e.getMessage());
+                    errorUsersMap.put(userName, e.getMessage());
                 }
             }
+        }
 
-            if (fail && success) {
-                throw new UserAdminException("Error occurs while importing user names. " +
-                        "Some user names were successfully imported. Some were not. Last error was : " + lastError);
-            }
+        String summeryLog = super.buildBulkImportSummary();
+        log.info(summeryLog);
 
-            if (fail && !success) {
-                throw new UserAdminException("Error occurs while importing user names. " +
-                        "All user names were not imported. Last error was : " + lastError);
-            }
-            if (isDuplicate) {
-                throw new UserAdminException("Detected duplicate user names. " +
-                        "Failed to import duplicate users. Non-duplicate user names were successfully imported.");
-            }
-        } catch (UserAdminException e) {
-            throw e;
+        JSONConverter jsonConverter = new JSONConverter();
```

Comment:
Shall we change this logic, by having AuditLog class extends from Log. then do this task within the new method 
audit(params). So that the code is better encapsulated and reusable.

Reply:
I think this would be a kernel level fix to the actual audit log class?